src/app/view/components/_common/push-publish-dialog/push-publish-dialog.component.ts

```diff
@@ -0,0 +1,121 @@
+import { BehaviorSubject } from 'rxjs/Rx';
+import { Component, Input, Output, EventEmitter, ViewEncapsulation } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { OnInit } from '@angular/core/src/metadata/lifecycle_hooks';
+import { PushPublishService } from '../../../../api/services/push-publish/push-publish.service';
+import { SelectItem } from 'primeng/primeng';
+import { DotMessageService } from '../../../../api/services/dot-messages-service';
+
+@Component({
+    encapsulation: ViewEncapsulation.None,
+    selector: 'dot-push-publish-dialog',
+    styleUrls: ['./push-publish-dialog.component.scss'],
+    templateUrl: 'push-publish-dialog.component.html'
+})
+
+export class PushPublishContentTypesDialogComponent implements OnInit {
+    publishdate: Date;
+    expiredate: Date;
+    form: FormGroup;
+    errorMessage: BehaviorSubject<string> = new BehaviorSubject(null);
+    forcePushCheckbox = false;
+    pushActions: SelectItem[];
+    publish = true;
+    showExpireDate = false;
+    submitted = false;
+    @Input() displayDialog = false;
+    @Input() contentTypeId: string;
+    @Output() cancel = new EventEmitter<boolean>();
+
+    constructor(
+        private pushPublishService: PushPublishService,
+        public fb: FormBuilder,
+        public dotMessageService: DotMessageService
+    ) {}
+
+    ngOnInit() {
+        this.pushActions = [
+            { label: 'Push', value: 'publish' },
+            { label: 'Remove', value: 'expire' },
+            { label: 'Push Expire', value: 'publishexpire' }
+        ];
+        this.publishdate = new Date;
+        this.expiredate = new Date;
+
+        this.dotMessageService.getMessages([
+            'contenttypes.content.push_publish',
+            'contenttypes.content.push_publish.I_want_To',
+            'contenttypes.content.push_publish.force_push',
+            'contenttypes.content.push_publish.publish_date',
+            'contenttypes.content.push_publish.expire_date',
+            'contenttypes.content.push_publish.push_to',
+            'contenttypes.content.push_publish.push_to_errormsg',
+            'contenttypes.content.push_publish.form.cancel',
+            'contenttypes.content.push_publish.form.push'
+        ]).subscribe();
+
+        this.form = this.fb.group({
+            pushActionSelected: [this.pushActions[0].value || '', [Validators.required]],
+            publishdate: [this.publishdate || '', [Validators.required]],
+            expiredate: [this.expiredate || '', [Validators.required]],
+            environment: ['', [Validators.required]],
+            forcePush: this.forcePushCheckbox || ''
```

Comment:
I think yo dont need any of this: this.publishdate, this.expiredate, this.forcePushCheckbox